src/System.Text.Json/tests/Serialization/UserReports.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Text.Json.Serialization;
+using Xunit;
+
+namespace System.Text.Json.Tests.Serialization
+{
+    /// <summary>
+    /// Catch-all location for general user reported scenarios that aren't specific to one serialization feature.
+    /// </summary>
+    public class UserReports
+    {
+        [Fact]
+        public void Issue38568()
+        {
+            // https://github.com/dotnet/corefx/issues/38568
```

Comment:
I don't think we should be naming/classifying tests like this.  In a few months it won't matter why we added this test, whether it was because we thought to initially or because someone else found it, and "Issue38568" does nothing to describe what this test is validating, and storing it in UserReports means it's not in the same place as the rest of the tests testing related functionality.

Reply:
It's validating a code sample we're giving to a customer reported issue. There is no specific bug, and it crosses multiple features. Picking one area bucket seems a bit arbitrary. I don't want it to lose the original reported scenario in the future by not tracking that it is replicating a specific situation (e.g. someone changes the test).

It's a scenario test, we can call it `CustomerScenario_38568` or is there another name you'd prefer?

Reply:
> There is no specific bug, and it crosses multiple features.

What specifically is it validating?  We're not going to add to our test suite every piece of code that's ever written using these APIs.  What is this one testing that our other tests don't?  It should be named based on that.

Reply:
> What specifically is it validating?

A code sample we're giving to a customer in the linked issue.

> We're not going to add to our test suite every piece of code that's ever written using these APIs.

I'm not suggesting we do. But we should have representative "real world" scenarios and we should be validating at least a subset of examples we give out publicly via GitHub issues or our documentation.

Reply:
Then the tests should be named on what's being tested, what the "real world" scenario is accomplishing, not an arbitrary effectively randomly generated five digit number that's meaningless unless you go and read pages of GitHub text to figure out why the rest is meaningful.

Reply:
Would `ReadingObjectWithUncStringEnumAndCustomDateTimeConverter` be ok? I didn't start there as it didn't communicate what I was trying to do, which is test a real reported scenario. From my perspective that detail was secondary.

Reply:
I think regression tests are valuable.  Often times we miss things because we don't test how multiple features work together.  It's also valuable to distill real customer usage into a test case.  
I do think a more meaningful name is helpful.  Issue links are good, but better for the comments.  One thing to remember in the naming is that this is the first thing you see when the test fails, so having some bit of useful information there can help easily grok what might have gone wrong when a test fails and save having to actually dig into the code or go look at the issue link.

Reply:
Does this test even fail without this change? If not, it seems weird to add it in the same PR.  I'd expect to see tests that show that we better roundtrip URIs that have lossy/destructive transformations through ToString(), so that if just the functional line of code was reverted there'd be a test that failed.

As for the name/value of this test; I agree that it needs a name descriptive of its purpose, not its origin (mainly for the ease of linking to a person changing code that yes, their code probably impacted the scenario for the test).

`DeserializeMixingCustomAndSystemConverters`, perhaps?

Reply:
> Does this test even fail without this change?

No, and I agree that it is a little strange. The user scenario caused me to look deeper at the Uri stuff which uncovered the other issue that the user scenario didn't explicitly cover.

>  I'd expect to see tests that show that we better roundtrip URIs that have lossy/destructive transformations through ToString()

I did add some of those specifically.

Reply:
Changed names and added more explicit comments